### ISSUE TITLE
Script removes task information from filenames.

### ISSUE THREAD

**Issue by juliam98** [UNFmontreal/Dcm2Bids#298]
I get a warning that `task` information is not found for datatype 'func' and suffix 'bold'. This does not seem correct according to the [BIDS docs](https://bids-specification.readthedocs.io/en/stable/modality-specific-files/magnetic-resonance-imaging-data.html#task-including-resting-state-imaging-data) which list the suffix `bold` as one of the options for `func` data and the template does show `task-<label>` entity. 

```
WARNING | {'task'} have not been found for datatype 'func' and suffix 'bold'.
WARNING | ✅ Filename was reordered according to BIDS entity table order:
                from:   sub-207_ses-MRI1_task-working-memory_bold
                to:     sub-207_ses-MRI1_bold
INFO    | sub-207_ses-MRI1_bold  <-  024_scans_BOLD_MB8_SWM_20231026131547
WARNING | {'task'} have not been found for datatype 'func' and suffix 'bold'.
WARNING | ✅ Filename was reordered according to BIDS entity table order:
                from:   sub-207_ses-MRI1_task-working-memory_bold
                to:     sub-207_ses-MRI1_bold
INFO    | sub-207_ses-MRI1_bold  <-  025_scans_BOLD_MB8_SWM_MoCo_20231026131547
```
and 

```
WARNING | ✅ Filename was reordered according to BIDS entity table order:
                from:   sub-207_ses-MRI1_task-working-memory_run-01_bold
                to:     sub-207_ses-MRI1_run-01_bold
WARNING | ✅ Filename was reordered according to BIDS entity table order:
                from:   sub-207_ses-MRI1_task-working-memory_run-02_bold
                to:     sub-207_ses-MRI1_run-02_bold
```

The output is `sub-207_ses-MRI1_run-01_bold.nii.gz` and `sub-207_ses-MRI1_run-02_bold.nii.gz` - so the task information was completely removed.

The config file I provided distinguished those two files by the property [rec](https://bids-specification.readthedocs.io/en/stable/appendices/entities.html#rec) to indicate that one version is motion corrected. This was also removed by `dcm2bids`.

I am not sure if I'm doing something wrong here or if this is a mistake in the script. Would appreciate some advice. Thanks!

**Comment by arnaudbore** [comment]
Hi @juliam98 ,
You task name is not BIDS compliant because of the "-" in the task name (-working-memory).  
Try to use task-workingMemory it should work. 
Let me know
Arnaud

**Comment by juliam98** [comment]
Hi Arnaud,

thanks, I have corrected that now to `workingmemory`, but overall I expected to get 6 files (tasks: `rest`, `workingmemory`, and `movie`; each with and without `MoCo`). The other task names in the config.json file did not have a "-". I keep getting the same output:

workingmemory:
```
WARNING | {'task'} have not been found for datatype 'func' and suffix 'bold'.
INFO    | sub-207_ses-MRI1_bold  <-  024_scans_BOLD_MB8_SWM_20231026131547
WARNING | {'task'} have not been found for datatype 'func' and suffix 'bold'.
INFO    | sub-207_ses-MRI1_bold  <-  025_scans_BOLD_MB8_SWM_MoCo_20231026131547
```
movies:
```
WARNING | {'task'} have not been found for datatype 'func' and suffix 'bold'.
INFO    | sub-207_ses-MRI1_bold  <-  037_scans_BOLD_MB8_Movies_20231026131547
WARNING | {'task'} have not been found for datatype 'func' and suffix 'bold'.
INFO    | sub-207_ses-MRI1_bold  <-  038_scans_BOLD_MB8_Movies_MoCo_20231026131547
```
rest:
```
WARNING | {'task'} have not been found for datatype 'func' and suffix 'bold'.
INFO    | sub-207_ses-MRI1_bold  <-  039_scans_BOLD_MB8_Resting_State_20231026131547
WARNING | {'task'} have not been found for datatype 'func' and suffix 'bold'.
INFO    | sub-207_ses-MRI1_bold  <-  040_scans_BOLD_MB8_Resting_State_MoCo_20231026131547
```

Weirdly the output is still `sub-207_ses-MRI1_run-01_bold.nii.gz` and `sub-207_ses-MRI1_run-02_bold.nii.gz`, so one pair of func task scans did get moved to the BIDS folder, when I checked the sidecar file it seems it's the workingmemory task, I thought it's because the names end up the same so they get overwritten, but the last pair (chronologically) is `rest` so that should be the remaining two. 

I initially had `"custom_entities": "task-[X]",` in the config.json file for all 6 files, which is probably why it appeared in the output at all. I deleted it from all except the two `rest` files, and now the output filenames don't mention it for any of the tasks. I have a feeling my `config.json` file might be a problem but I am not sure why.

**Comment by arnaudbore** [comment]
Can you share your config.json file ? You definitively need to add custom_entities for each fMRI so your bold data are bids compliant.

**Comment by juliam98** [comment]
Hi, I attached the [config.json](https://github.com/user-attachments/files/15512954/config.json) file. I put back `custom_entities: task-[name]` in the file and re-run `dcm2bids`.



I am also attaching the output I get in the terminal (with all the directories and files created)
[BIDS_output.txt](https://github.com/user-attachments/files/15513094/BIDS_output.txt)

I suspect the other values that were not added to the filename such as `"rec": "MoCo"` should have been added as custom_entities too?

**Comment by arnaudbore** [comment]
Hi @juliam98 ,

There are a couple of issues in your config file. There issues can easily fix by following the example here: https://unfmontreal.github.io/Dcm2Bids/3.1.1/how-to/create-config-file/
- Please use only the key allowed in a config file (L5: seq, L7: seg, L31: desc, L141: run, L556: rec, L557: task, ... ) all these keys are not allowed and dcm2bids ends up failing at different steps.
- There is no need for an empty sidecar_changes key

Like I said take some time to carefully read the documentation it will help you understand your issues.
If you still have some issues don't hesitate to ask.

Hope it helps 
Arnaud

**Comment by juliam98** [comment]
Thanks for specifying, It wasn't clear which keys are allowed in the config file, I have now changed the config file so it includes only `datatype`, `suffix`, and `custom_entities`. The files appear to be named correctly, however I do still get a `WARNING` about the entties I provided in `custom_entities` not being valid BIDS entities. I have copied the log just FYI. [sub-207_ses-MRI1_20240605-135304.log](https://github.com/user-attachments/files/15615376/sub-207_ses-MRI1_20240605-135304.log)
